### PR TITLE
(PUP-8585) Error mgs displayed for device resource

### PR DESCRIPTION
**Pull request by tphoney**
Print error messages for puppet device --resource. Previously --debug needed to be set as well

### CODE REVIEW

**Comment by shermdog** [comment]
:+1:

**Comment by tphoney** [comment]
@joshcooper can you have a look, thanks in advance

**Comment by tphoney** [comment]
@davidmalloncares

File: lib/puppet/application/device.rb [hunk 1]
```diff
@@ -352,27 +344,31 @@ def setup_host
   end
 
   def setup
-    setup_logs
+    if options[:resource]
+      Puppet.settings.use :main, :agent, :ssl
```
Comment (joshcooper) [inline lib/puppet/application/device.rb]: why is `:device` excluded?

**Comment by tphoney** [comment]
Thanks for the review @joshcooper . Commit is amended with the extra explanation.

The reason :device is ommited is because this affects what messages get displayed. Previously device was meant to be run in a headless mode, ie no messages displayed when run. 
Now that we have --apply and --resource (interactive) we want messages displayed on the cli.

**Comment by davidmalloncares** [comment]
Hi guys, this is one we'd like to get in before the May 21st Stop Ship date for Agent changes - @joshcooper - if you are good with the changes are we ok to merge?

**Comment by davidmalloncares** [comment]
Hi @joshcooper , I believe the master branch is open again for PRs to be merged - are you ok with the PR being merged?

**Pull request merged**